src/sentry/static/sentry/app/views/settings/organizationFiltersAndSampling/modals/transactionRuleModal.tsx

```diff
@@ -0,0 +1,164 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+import {ModalRenderProps} from 'app/actionCreators/modal';
+import CheckboxFancy from 'app/components/checkboxFancy/checkboxFancy';
+import ExternalLink from 'app/components/links/externalLink';
+import {t, tct} from 'app/locale';
+import space from 'app/styles/space';
+import {Organization} from 'app/types';
+import {
+  DynamicSamplingConditionOperator,
+  DynamicSamplingRule,
+} from 'app/types/dynamicSampling';
+import {defined} from 'app/utils';
+import Field from 'app/views/settings/components/forms/field';
+import Form from 'app/views/settings/components/forms/form';
+import NumberField from 'app/views/settings/components/forms/numberField';
+import SelectField from 'app/views/settings/components/forms/selectField';
+
+import MatchField from './matchField';
+
+const conditionChoices = [
+  [DynamicSamplingConditionOperator.ALL, t('All Transactions')],
+  [DynamicSamplingConditionOperator.GLOB_MATCH, t('Releases')],
+  [DynamicSamplingConditionOperator.STR_EQUAL_NO_CASE, t('Enviroments')],
+  [DynamicSamplingConditionOperator.EQUAL, t('Users')],
+];
+
+type Props = ModalRenderProps & {
+  organization: Organization;
+  onSubmit: (rule: DynamicSamplingRule) => void;
+};
+
+type State = {
+  tracing: boolean;
+  condition: DynamicSamplingConditionOperator;
+  match: string;
+  sampleRate?: number;
+};
+
+class TransactionRuleModal extends React.Component<Props, State> {
+  state: State = {
+    tracing: true,
+    condition: DynamicSamplingConditionOperator.ALL,
+    match: '',
+  };
+
+  handleSubmit = async () => {
+    const {sampleRate} = this.state;
+
+    if (!defined(sampleRate)) {
+      return;
+    }
+
+    // TODO(PRISCILA): Finalize this logic according to the new structure
+  };
+
+  handleSubmitSuccess = () => {};
+
+  handleClickTracing = () => {
+    this.setState(prevState => ({tracing: !prevState.tracing}));
+  };
+
+  handleChange = <T extends keyof State>(field: keyof State, value: State[T]) => {
+    if (field === 'sampleRate') {
+      this.setState(prevState => ({
+        ...prevState,
+        sampleRate: value ? Number(value) : undefined,
+      }));
+      return;
+    }
+    this.setState(prevState => ({...prevState, [field]: value}));
```

Comment:
Do you need the prevState here?

Reply:
yes I do 😉

![image](https://user-images.githubusercontent.com/29228205/105204916-ef8f9400-5b44-11eb-9277-ef3b1a69a3cf.png)